src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs

```diff
@@ -1,24 +1,13 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
-using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Routing
 {
     public abstract class LinkGenerator
     {
-        public abstract bool TryGetLink(
-            HttpContext httpContext,
-            IEnumerable<Endpoint> endpoints,
-            RouteValueDictionary explicitValues,
-            RouteValueDictionary ambientValues,
-            out string link);
+        public abstract bool TryGetLink(LinkGeneratorContext context, out string link);
```

Comment:
👍 I'm pro a context object. More extensible, and there were getting too many parameters

Reply:
I'm ok with making some tenative decisions here if we think it's an improvement. We still need to do design work on this area in preview1 - so I would suggest not making any expensive changes until we've had a proper design review.